source/heating_model/compositional_heating.cc

```diff
@@ -136,16 +89,23 @@ namespace aspect
       {
         prm.enter_subsection("Compositional heating");
         {
-
-          field_used_in_heat_production_averaging = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_int(Utilities::split_string_list(prm.get("Use compositional field for heat production averaging"))),
-                                                    n_fields,
-                                                    "Use compositional field for heat production averaging");
-
-          heating_values = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Compositional heating values"))),
-                                                                   n_fields,
-                                                                   "Compositional heating values");
-
-
+          std::vector<int> used_fields = Utilities::possibly_extend_from_1_to_N (
+                                           Utilities::string_to_int(
+                                             Utilities::split_string_list(
+                                               prm.get("Use compositional field for heat production averaging"))),
+                                           n_fields,
+                                           "Use compositional field for heat production averaging");
+
+          fields_used_in_heat_production_averaging.resize(used_fields.size());
+          for (unsigned int i=0; i<used_fields.size(); ++i)
+            fields_used_in_heat_production_averaging[i] = used_fields[i];
```

Comment:
Is this correct? The right hand side are integers, the left are booleans. If this is correct, then I misunderstood what `used_fields` contained before. Can you make the cast explicit by putting a boolean expression on the rhs?